javascript/ql/src/Security/CWE-020/IncompleteUrlRegExp.ql

```diff
@@ -0,0 +1,70 @@
+/**
+ * @name Incomplete URL regular expression
+ * @description Security checks on URLs using regular expressions are sometimes vulnerable to bypassing.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id js/incomplete-url-regexp
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-20
+ */
+
+import javascript
+import semmle.javascript.security.dataflow.RegExpInjection
```

Comment:
This may be part of the performance issue. By bringing this taint tracking-configuration into scope, it will be evaluated alongside `IncompleteUrlRegExpTracking::Configuration` in sort of a "get 1 pay for 2" deal. Or at least I think it will (@xiemaisi?)

I think it would be better to extract the relevant bits of `RegExpInjection::Sink` into a shared library, and then also create a `.qll` file for this library, like we have for the other taint-tracking queries.